src/Corax/Mappings/IndexFieldsMappingBuilder.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Sparrow.Server;
+using Sparrow.Threading;
+using Voron;
+
+namespace Corax.Mappings;
+
+public class IndexFieldsMappingBuilder : IDisposable
+{
+    private readonly ByteStringContext _context;
+    private const short LongSuffix = 19501; //"-L"
+    private const short DoubleSuffix = 17453; //"-D"
+    
+    private readonly Dictionary<Slice, IndexFieldBinding> _fields;
+    private readonly Dictionary<int, IndexFieldBinding> _fieldsById;
+    
+    private bool _materialized;
+    private readonly bool _isDynamic;
+    
+    private Func<string, Analyzer> _exactAnalyzer;
+    private Func<string, Analyzer> _searchAnalyzer;
+    private Analyzer _defaultAnalyzer;
+    private readonly bool _isForWriter;
+    private int _maximumOutputSize = Constants.Terms.MaxLength;
+    private int _maximumTokenSize = Constants.Terms.MaxLength;
+
+    public int Count => _fields.Count;
+    
+    public IndexFieldsMappingBuilder(bool isForWriter, bool isDynamic = false)
```

Comment:
I wonder if the code wouldn't be more readable if introduced dedicated methods with meaningful names to create an instance of the builder instead of passing boolean parameters to the constructor. I mean the following changes:

1. Make constructor private
2. Create the following methods:
```
public static IndexFieldsMappingBuilder CreateForWriter(bool supportsDynamicFields = false)
{
    return new IndexFieldsMappingBuilder(true, supportsDynamicFields)
}
```

```
public static IndexFieldsMappingBuilder CreateForSearcher()
{
    return new IndexFieldsMappingBuilder(false, false)
}
```